Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java

```diff
@@ -174,6 +179,7 @@ private MapPolyline parseLineString()
     MapPolyline line = mFactory.createMapPolyline();
     boolean hasParsedCoordinates = false;
     while (mParser.next() != XmlPullParser.END_TAG) {
+      verifyNotEndOfDocument();
```

Comment:
This pattern seems fragile as now you need to keep putting this every time to move to the next element. A function to move to next and verify not end would fix this copy/paste problem as well as help future proof the code.